src/Service/Project/ChartManager.php

```diff
@@ -127,6 +127,42 @@ public function getGantt(Etude $etude, $type)
                 ];
             }
             $data = $dataSauv;
+            if ($etude->getCe() && $etude->getCe()->getDateSignature()) {
+                $date = $etude->getCe()->getDateSignature();
+                if ($naissance >= $date) {
+                    $naissance = clone $date;
+                }
+                if ($mort <= $date) {
+                    $mort = clone $date;
+                }
+
+                $data[] = ['x' => count($cats), 'y' => $date->getTimestamp() * 1000,
+                           'titre' => 'Convention d\'étude', 'detail' => 'signé le ' . $date->format('d/m/Y'),
+                ];
+                $series[] = ['type' => 'scatter', 'data' => $data,
+                             'marker' => ['symbol' => 'triangle', 'fillColor' => 'red'],
+                ];
+            }
+            $data = $dataSauv;
+            if ($etude->getPvis()) {
+                foreach ($etude->getPvis() as $pvi) { 
+                    if ($pvi->getDateSignature()){
+                        $date = $pvi->getDateSignature();
```

Comment:
Pour des raisons de lisibilité, on préfère habituellement retourner immédiatement en cas de condition invalide plutôt que faire pleins de if (car si plusieurs conditions à vérifier, ça fait beaucoup d'imbrications).

```suggestion
                foreach ($etude->getPvis() as $pvi) { 
                    if (!$pvi->getDateSignature()){
                    continue;
                   }
                        $date = $pvi->getDateSignature();
```